### PR TITLE
Frontend 1882 - Add Static Side Nav with Main Sections

### PR DESCRIPTION
**Pull request by langlk**
Resolves [FRONTEND-1882](https://algorithmia.atlassian.net/browse/FRONTEND-1882)

**Notes:**
* This relies on styles in https://github.com/algorithmiaio/synapse/pull/188
* You *will* need to rebuild the synapse dist files locally before this will look alright
* This doesn't have any javascript, so it will only change when you actually visit a page.
* There aren't any icons for the main menu items like in the spec: I'm going to add those when I add material icons to the whole project.

### CODE REVIEW

File: _data/menus.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+main:
```
Comment (langlk) [inline _data/menus.yml]: @peckjon I wanted to get your input on whether this is a reasonable setup for you to use: I've added [jekyll-menus](https://github.com/forestryio/jekyll-menus) to handle menu data.

In this file,`main` lists the root menu items (just the dev center, api docs, and a link to algorithmia.com), and each item that has a submenu gets an identifier. (explanatory comment 1/2)

File: _pages/algorithm-development.md [hunk 2]
```diff
@@ -6,6 +6,11 @@ excerpt: "Basic Guides for Algorithm Developers"
 show_related: false
 redirect_from:
   - /algorithm-development/basic-guides/
+menus:
+  devcenter:
```
Comment (langlk) [inline _pages/algorithm-development.md]: (explanatory comment 2/2)
Then each page you want to appear as an item in a submenu gets this `menus` section in its front matter, in the form:
```yaml
[menu identifier]:
  url: "what url the menu link should have"
  title: "what the menu link should say"
  weight: "where in menu it should appear, starting at 1"
```

**Comment by peckjon** [comment]
@langlk I like this pattern, and retaining frontmatter will improve future integrations with 3rd party tools. Thx!

**Comment by bnlb** [comment]
Really excited to see this come together. 🤸‍♂

**Pull request merged**